packages/stryker/test/unit/reporters/ci/CircleProviderSpec.ts

```diff
@@ -0,0 +1,75 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import * as environmentVariables from '../../../../src/utils/objectUtils';
+
+import CircleProvider from '../../../../src/reporters/ci/CircleProvider';
+
+describe('CircleCI Provider', () => {
+  let getEnvironmentVariables: sinon.SinonStub;
+
+  const circleProvider = new CircleProvider();
```

Comment:
I know the `CircleProvider` does not have state. Yet I would feel better if we created a new one for each test. If it would have state in the future, you wouldn't automatically expect state to flow from test to test.

Reply:
In fact, I thought of it the other way around, in a more test-driven approach: I don't *want* the Providers to have state. So let's set it up in a way that keeping state would be hard to achieve :-).

Reply:
This suggests that the test would be red, while in fact it could just as easily be that the tests are green for 1 test run, but red the next one. You'd never notice, because tests always run just once, *unless* of course during mutation testing. The result is a very high mutation score, seeing as all runs fail. We've been burned by this once before, and it took me hours to figure out (why do we have a mutation score of 98%?? surely our tests aren't *that* good)

I agree that it is a bad practice to keep state in a provider, but on the other hand I think it is a bad practice to share state over tests. Both should stand on its own. Do you agree?